Fix non localized numerals

Kalovelo opened the pull request:
### Summary
Non translated numerals have been fixed in the below occurrences:
- Exam question counter in list 
- Exam question counter in title
- Exam answer counter
- Channel's version
- User's Birth year on profile page and other places
- Time elapsed on `StatusElapsedTime` componenent
- Total learners in group (used by coach)

_Note: numeral formatting on kolibri design system componenets such as `KTextBox` couldn't be fixed since that is a design system's issue_



### Reviewer guidance
The reviewer can visit the below pages
- User profile
- Device channels
- Exam (as a learner)
- As a coach, plan a class and browse the groups


### References
This PR resolves #7696 .


----

### Contributor Checklist


PR process:

- [x] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

jonboiser commented:
@Kalovelo Could you rebase this on `learningequality/develop`, and remove any commits that aren't yours? I already changed the target of this PR.

Kalovelo commented:
Rebased :heavy_check_mark:

jonboiser commented on kolibri/locale/mr/LC_MESSAGES/kolibri.plugins.learn.app-messages.json:
```diff
@@ -1,6 +1,6 @@
 {
   "AllClassesPage.noClasses": "तुम्हाला कोणताही वर्ग नेमून दिलेला नाही",
-  "AnswerHistory.question": "प्रश्न { num }",
+  "AnswerHistory.question": "प्रश्न xxx  {num, number, integer}",
```
@radinamatic this looks a little weird. is this "xxx" actually part of the localization?

Kalovelo replied:
@jonboiser  this is my bad, I added some test strings to test on hot reload, fixing it :cry:

jonboiser commented on kolibri/plugins/learn/assets/src/views/ExamPage/AnswerHistory.vue:
```diff
@@ -91,7 +91,7 @@
     },
     $trs: {
-      question: 'Question { num }',
+      question: 'Question { num, number, integer}',
```
```suggestion
      question: 'Question { num, number, integer }',
```

jonboiser replied:
Did you modify the locale JSON file directly? I see changes to other messages like
```
  "ExamPage.question": "Câu hỏi {num, number, integer} trong số {total, number, integer}",
```

but don't see `ExamPage.vue` in this diff. Same with `ChannelVersion`. 

Normally, the changes to the locale JSON files are something we handle internally because they require uploading data to the translation platform! What you should do is actually update the Vue files like you did here, and we will handle updating the JSON files later. Sorry about that!

Here is some docs about the whole process https://kolibri-dev.readthedocs.io/en/develop/i18n.html

If you only update the JSON files, but not the Vue components, your changes will be undone the next time we download translations from the translation platform.😞

Kalovelo replied:
Thank you @jonboiser , and sorry about that. I will set the fixes only on `.vue` components and update the PR

Kalovelo committed:
refactor: set makeYearOptions as componenent function in order to use i18n

Kalovelo committed:
localize: fix non-localized numerals used in components

radinamatic commented:
@Kalovelo Thank you for working on this! 👏🏽 💯 

I know it's still in progress, but here's what I can still notice when I run the server in development from your branch:

Location | Screenshot
--- | ---
**Device > Channels**<br><br> Channel version seems localized in Chrome, but not in Firefox. | ![LEDev2104 (start)  Running  - Oracle VM VirtualBox_011](https://user-images.githubusercontent.com/1457929/113636305-924ead80-9672-11eb-90e3-037d76840d54.png)
**Facility > Classes**<br><br> Number of learners in a class not localized in both Chrome and FF. | ![LEDev2104 (start)  Running  - Oracle VM VirtualBox_012](https://user-images.githubusercontent.com/1457929/113636717-536d2780-9673-11eb-8a2c-0ae8d1bee2a5.png)
**Coach > Plan > Create exam**<br><br> Number of questions input field fails in Chrome, but not in Firefox (go figure 😕). | ![LEDev2104 (start)  Running  - Oracle VM VirtualBox_013](https://user-images.githubusercontent.com/1457929/113636938-c4acda80-9673-11eb-9d24-215b29fcd6be.png)
**Learn > Class > Exam**<br><br> Question numbers not localized in both Chrome and FF. | ![LEDev2104 (start)  Running  - Oracle VM VirtualBox_014](https://user-images.githubusercontent.com/1457929/113637054-fc1b8700-9673-11eb-809a-08396ffe0431.png)

jonboiser commented:
@radinamatic Those specific spots won't be localized until we get new translations from crowdin. Originally, @Kalovelo did update the locale files (i.e. self-translated them!), but undid those changes after I explained our localization processes. I'm confident that the numbers in those messages will be localized properly once  we get the updated translations back, since they will have the full `{num, number, integer}` annotation.

There some places, like the tables in the coach reports, where the numbers are localized since they don't need to be pre-translated.

radinamatic commented:
Awesome then, more progress then what I can see! 👍🏽

jonboiser commented:
I think my missing translations explanation accounts for screenshots 1 and 4, but not the others. Apostolos, did mention an issue with localizing the `KTextbox` character counter (a kolibri-design-system issue), which might account for screenshot 3.

Screenshot 2 does look like a potential spot where we did not pass the number into `$formatNumber` or something else is going on.

jonboiser commented:
@Kalovelo It looks like this one number needs to be passed to `$formatNumber`

https://github.com/learningequality/kolibri/blob/4b2e9cc8ee8a562021e5bd363c64f606f47e8869/kolibri/plugins/facility/assets/src/views/ManageClassPage/index.vue#L79

Actually, would you mind replacing these lines like this one that use `coachStrings/integer` with `$formatNumber` as well? This will allow us to skip an unnecessary translation step for these numbers.

https://github.com/learningequality/kolibri/blob/4b2e9cc8ee8a562021e5bd363c64f606f47e8869/kolibri/plugins/coach/assets/src/views/CoachClassListPage.vue#L59

I'm looking through the other components with tables to see if there's any other spots we've missed.

Kalovelo committed:
localize(ManageClassPage): add numeral format on learner count

Kalovelo committed:
refactor(localization): use formatNumber in components instead of integer translation string

Kalovelo commented:
I added the missing numeral formatting on screenshot 2 coach's classes component and refactored `commonCoachStrings` integer formatting by removing it from `commonCoachStrings` and setting `$formatNumber` in each component occurence in order to skip unnecessary translations

jonboiser reviewed (approved):
Looks good! Thank you!
